datasets/pcfg/tensorflow-datasets/pcfg_productivity_data.py

```diff
@@ -0,0 +1,87 @@
+"""pcfg_productivity_data dataset."""
```

Comment:
It seems to me that the two dataset classes only differ in a few details (a couple of paths/filenames), right? Wouldn't it be better to then define a PcfgData class that has parameters in the constructor? and then the other datasets only require redefining the default constructor to call the PcfgData constructor with the right arguments, right? that would save 100s of lines of code. Let me know if it does not make sense :)

Reply:
Because I am not sure if there is anyway to pass constructor attributes to the tfds command, I have taken a different approach by updating pcfg_productivity_data.py to generate both the original and the iterative decoding data. After running $ tfds build pcfg_productivity_data.py to build the dataset, the original data can be found in the 'train' and 'test' splits; and the iterative decoding data, in the 'it_dec_train', 'it_dec_val' and 'it_dec_test' splits. Per Josh's suggestion, the 'it_dec_val' set is used to assess the ability of the model to generalize in the standard sense (it has the same form as the 'it_dec_train' set); the 'it_dec_test' set is used to assess the ability of the model to generalize through iterative decoding. Let me know what you think!